File: bubble sort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {

    //Declara as variaveis, cria uma seed, e cria o array
    int i;
    srand(time(NULL));
    int tamanho = 100;
    int nums[tamanho];

    // Poe numeros aleatorios no array
    for (i = 0; i < tamanho; i++) {
        nums[i] = rand() % tamanho;
    }

    //Cria a variavel que vai ver se o codigo repetira
    int moveu;

    do {
        // i é a variável de controle do loop, caso ela esteja na ultima iteração, ao invés de continuar o loop, 
        // irá verificar se há elementos sendo movidos no array, em caso negativo,
        // ele termina o loop. Caso haja movimentos, ele continua o loop, recomeçando ele
        moveu = 0;
        for (i = 0; i < tamanho - 1; i++) { 
            if (nums[i] > nums[i + 1]) {
                int menor = nums[i + 1];
                nums[i + 1] = nums[i];
                nums[i] = menor;
                moveu = 1;
            }
        }
        //imprime
            for (i = 0; i < tamanho; i++) {
                printf("%d ", nums[i]);
            }

    // Encerra o código
    } while (moveu == 1);
    // Quebra a linha
    printf("\n");
    return 0;
}

File: bubble sort.py
lista = [5, 3, 4, 0, 9, 1, 2]
moveu = False
i = 0

while True:
    
    # verifica se está no ultimo elemento da lista
    if i+1 == len(lista):
        print(lista)
        # reseta a variável i para reiniciar caso tenha movido, caso não tenha movido ele encerra o loop
        i = 0
        if not moveu:
            break
        else:
            moveu = False
            continue
    # verifica se o elemento da esquerda é menor que o da direita
    elif lista[i] > lista[i+1]:
        # caso seja, troca de lugar o elemento da esquerda com o da direita
        menor = lista[i+1]
        lista[i+1] = lista[i]
        lista[i] = menor
        # sinaliza que moveu para poder continuar o loop
        moveu = True

    # incrementa 1 na variável i, que está sendo usada para percorrer todos os elementos da lista
    i += 1
    